packages/next/build/analysis/get-page-static-info.ts

```diff
@@ -326,15 +329,28 @@ export async function getPageStaticInfo(params: {
 
     const requiresServerRuntime = ssr || ssg || pageType === 'app'
 
-    resolvedRuntime =
-      SERVER_RUNTIME.edge === resolvedRuntime
-        ? SERVER_RUNTIME.edge
-        : requiresServerRuntime
-        ? resolvedRuntime || nextConfig.experimental?.runtime
-        : undefined
+    resolvedRuntime = isEdgeRuntime(resolvedRuntime)
+      ? resolvedRuntime
+      : requiresServerRuntime
+      ? resolvedRuntime || nextConfig.experimental?.runtime
+      : undefined
 
-    if (resolvedRuntime === SERVER_RUNTIME.edge) {
-      warnAboutExperimentalEdgeApiFunctions()
+    if (resolvedRuntime === SERVER_RUNTIME.experimentalEdge) {
+      warnAboutExperimentalEdge()
+    }
+
+    if (
+      resolvedRuntime === SERVER_RUNTIME.edge &&
+      pageType === 'pages' &&
+      page &&
+      !isAPIRoute(page.replace(/^\/pages\//, '/'))
+    ) {
+      const message = `Page ${page} provided runtime 'edge', the edge runtime for rendering is currently experimental. Use runtime 'experimental-edge' instead.`
+      Log.error(message)
+
+      if (!isDev) {
```

Comment:
Small todo for later.
```suggestion
      // TODO: replace process.exit(1) with throwing.
      if (!isDev) {
```